### ISSUE TITLE
bug(parser): failed to parse member expression inside decorators.

### ISSUE THREAD

**Issue by Boshen** [Boshen/oxc#191]
Found this from a real project.

```
@decorator((baz) => ({
  bar: baz['']
}))
class Foo {}
```

```
  × Expect token
   ╭─[test.ts:1:1]
 1 │ @decorator((baz) => ({
 2 │   bar: baz['']
   ·           ┬
   ·           ╰── Expect `,` here, but found `[`
 3 │ }))
   ╰────
```

**Comment by Boshen** [comment]
Something's broken with the parser context. The function callback should probably fix the context here, but I wonder where's the grammar for this 🤔 

https://github.com/Boshen/oxc/blob/6a89f085b7bd993164efd5a6dcaf09b88d7dd593/crates/oxc_parser/src/js/expression.rs#L519

**Comment by YangchenYe323** [comment]
I noticed the grammar for a decorator (https://arai-a.github.io/ecma262-compare/snapshot.html?pr=2417#prod-Decorator) does not align much with normal expression. For example, the call can be only one level deep. I tried to use separate productions for decorators and get rid of the context.
<img width="755" alt="Screen Shot 2023-03-17 at 2 47 31 PM" src="https://user-images.githubusercontent.com/42916814/226013345-f73812a3-1ab5-4188-8d25-ee44913d0a73.png">
